problem7/src/main/java/problem7/CustomLinkedList.java

```diff
@@ -0,0 +1,109 @@
+package problem7;
+
+/**
+ * Task: <br>
+ * Write code to remove duplicates from an unsorted linked list
+ * @author Ramiz
+ *
+ */
+public class CustomLinkedList {
+
+	public Node head;
+	
+	public CustomLinkedList (int data){
+		head = new Node (data);
+	}
+	
+	/**
+	 * Adds a new node with entered value
+	 * @param data - integer value to be added
+	 * @return reference to itself for convenient adding
+	 */
+	public CustomLinkedList add (int data){
+		Node current = this.head;
+		while (current.next != null){
+			current = current.next;
+		}
+		current.next = new Node(data);
+		return this;
+	}
+	
+	/**
+	 * Removes node by selected value.
+	 * @param data - integer value used as key to delete node
+	 */
+	public void removeNode (int data){
+		if (head.data == data){
+			this.head = head.next;
+		}
+		Node current = this.head;
+		while (current.next != null){
+			if (current.next.data == data){
+				current.next = current.next.next;
+				break;
+			}
+			current = current.next;
+		}
+	}
+	
+	/**
+	 * Removes node by selected index
+	 * @param index - integer value, index of node to be deleted
+	 */
+	public void removeNodeByIndex (int index){
```

Comment:
See, you have two methods quite similar by goal and implementation. Please avoid copypaste things and repsect DRY principle.
One way to solve this issue is to add one more private method remove(Node node) and reuse it in both cases. But that should be your desision.